cunumeric/deferred.py

```diff
@@ -442,26 +657,32 @@ def set_item(self, key, rhs):
         # Check to see if this is advanced indexing or not
         if is_advanced_indexing(key):
             # Create the indexing array
-            index_array = self._create_indexing_array(key)
-            if index_array.shape != rhs.shape:
-                raise ValueError(
-                    "Advanced indexing array does not match source shape"
-                )
-            if self.ndim != index_array.ndim:
-                raise NotImplementedError(
-                    "need support for indirect partitioning"
-                )
+            (
+                copy_needed,
+                lhs,
+                index_array,
+                self,
+            ) = self._create_indexing_array(key, True)
+            rhs = self.runtime.to_deferred_array(rhs)
+            if rhs.shape != index_array.shape:
+                rhs_tmp = rhs._broadcast(index_array.base.shape)
+                rhs_tmp, rhs = rhs._copy_store(rhs_tmp)
+            else:
+                if rhs.base.transformed:
+                    rhs, rhs_base = rhs._copy_store(rhs.base)
+                rhs = rhs.base
 
             copy = self.context.create_copy()
-
-            copy.add_input(rhs.base)
+            copy.add_input(rhs)
             copy.add_target_indirect(index_array.base)
-            copy.add_output(self.base)
-
-            copy.add_alignment(index_array.base, rhs.base)
-
+            copy.add_output(lhs.base)
             copy.execute()
 
+            # TODO this copy will be removed when affine copies are
+            # supported in Legion/Realm
+            if lhs is not self or self.base.transformed:
```

Comment:
Is the `or self.base.transformed` necessary? If I'm understanding the code correctly, if `_create_indexing_array` returns `lhs` same as `self` that means the store has no transforms.

Reply:
That is right, it is not necessary